master/buildbot/status/buildstep.py

```diff
@@ -69,7 +69,9 @@ def __init__(self, parent, master, step_number, step_type):
         self.step_number = step_number
         self.hidden = False
         self.logs = []
-        self.urls = {}
+        self.urls = []
```

Comment:
@BarryUnity you've changed type here from dictionary to array that causes issues down to the code, some of the usages still assume they get the dictionary and try to call its methods. Please check usages of the variable and test the changes locally. There are might be front-end issues caused by JSON changes, but there shouldn't be any backend issues(for example, when you were trying to run build).